craft_store/http_client.py

```diff
@@ -0,0 +1,152 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store HTTPClient."""
+
+import logging
+import os
+from typing import Dict, Optional
+
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPClient:
+    """Generic HTTP Client to communicate with Canonical's Developer Gateway.
+
+    This client has a requests like interface, it creates a requests.Session
+    on initialization to handle retries over HTTP and HTTPS requests.
+
+    The default number of retries is set to 5 and can be overridden through
```

Comment:
The code uses 10, not 5. I suggest using module-level constants for this value and the backoff one, then use that constant to build this docstring and as a default in the code, so the docs never lie.